test/unit/metrics/test_extra_metrics.py

```diff
@@ -1,102 +1,331 @@
 # Copyright (c) Microsoft Corporation and contributors.
 # Licensed under the MIT License.
 
+import numpy as np
+import pytest
+import sklearn.metrics as skm
+
 import fairlearn.metrics as metrics
+from fairlearn.metrics._extra_metrics import _get_labels_for_confusion_matrix
+
+# =============================================
+
+
+class TestGetLabelsForConfusionMatrix:
+    def test_smoke(self):
+        r0 = _get_labels_for_confusion_matrix([0, 1], None)
+        assert np.array_equal(r0, [0, 1])
+        r1 = _get_labels_for_confusion_matrix([-1, 1], None)
+        assert np.array_equal(r1, [-1, 1])
+
+    def test_smoke_numeric_pos_label(self):
+        r0 = _get_labels_for_confusion_matrix([0, 2], 2)
+        assert np.array_equal(r0, [0, 2])
+        r1 = _get_labels_for_confusion_matrix([0, 2], 0)
+        assert np.array_equal(r1, [2, 0])
+
+    def test_smoke_alpha_pos_label(self):
+        r0 = _get_labels_for_confusion_matrix(['a', 'b'], 'b')
+        assert np.array_equal(r0, ['a', 'b'])
+        r1 = _get_labels_for_confusion_matrix(['a', 'b'], 'a')
+        assert np.array_equal(r1, ['b', 'a'])
+
+    def test_single_value_numeric_no_pos_label(self):
+        r0 = _get_labels_for_confusion_matrix([0], None)
+        assert np.array_equal(r0, [0, 1])
+        r1 = _get_labels_for_confusion_matrix([-1], None)
+        assert np.array_equal(r1, [-1, 1])
+        r2 = _get_labels_for_confusion_matrix([1], None)
+        assert np.array_equal(r2, [None, 1])
+
+    def test_single_value_numeric_pos_label(self):
+        r0 = _get_labels_for_confusion_matrix([0], 3)
+        assert np.array_equal(r0, [0, 3])
+        r1 = _get_labels_for_confusion_matrix([0], 0)
+        assert np.array_equal(r1, [None, 0])
+
+    def test_single_value_alpha_pos_label(self):
+        r0 = _get_labels_for_confusion_matrix(['a'], 'a')
+        assert np.array_equal(r0, [None, 'a'])
+        r1 = _get_labels_for_confusion_matrix(['a'], 0)
+        assert np.array_equal(r1, ['a', 0])
+
+    def test_too_many_values(self):
+        expected_msg = "Must have no more than two unique y values"
+        expected_msg_no_pos = \
+            "If pos_label is not specified, values must be from {0, 1} or {-1, 1}"
+        with pytest.raises(ValueError) as e0:
+            _get_labels_for_confusion_matrix([0, 1, 2], None)
+        assert str(e0.value) in {expected_msg, expected_msg_no_pos}
```

Comment:
The point of checking for the message text is to make sure that we've hit the error condition we're checking for. Having multiple options negates that.

Reply:
there are two error conditions that apply!

Reply:
Certainly, but one or the other should appear deterministically. It's why my version of the checks was very flat - checking for errors early and raising immediately makes it easier to see where you're going to end up.

I can see that if we're looking at the test 'being a user' then this does make sense. I'll just add a note to the test before checkin.

Reply:
Thanks for clarifying here. My intention was to make this test a bit future-proof in case we tweak the flow in `_get_labels_for_confusion_matrix`.